### DISCUSSION TITLE
FadeAnimation takes twice as long as expected

### DISCUSSION THREAD

**Discussion by bijington** [CommunityToolkit/Maui#1231] [General]
While converting the rest of the Animations from XCT I spotted that our use of the `Length` (MCT) / `Duration` (XCT) properties do not behave as I expected. Let's just look at the `FadeAnimation` example for now...

Currently if I create a `FadeAnimation` with a `Length` of 300 I expect it to complete it's animation after 300ms however looking at the code it will take twice as long:

```csharp
public override async Task Animate(VisualElement view)
{
    ArgumentNullException.ThrowIfNull(view);

    var originalOpacity = view.Opacity;

    await view.FadeTo(Opacity, Length, Easing);
    await view.FadeTo(originalOpacity, Length, Easing);
}
```

I believe the code should read:

```csharp
    await view.FadeTo(Opacity, Length / 2, Easing);
    await view.FadeTo(originalOpacity, Length / 2, Easing);
```

If we do decide this is a bug then it is a breaking change to fix it because we would be doubling the speed of everyones fade animations. I assume this is a fairly underused feature of the toolkit given we only have 1 animation in so far and people haven't asked for more.

What are peoples thoughts on this?

**Comment by TheCodeTraveler** [comment]
Oooh! I totally agree. As a user, if I specified that the `Length` is 500ms, but then the animation takes 1000ms, I would be perplexed. Yea, We should fix this. This would certainly be a breaking change, but we could queue up `CommunityToolkit.Maui v6.0.0` to also include Vlad's breaking change to Popup on Windows #1086.

The option for implementing breaking changes would be when we bump to .NET 8.0, but that won't be until November.

**Reply by TheCodeTraveler** [reply]
Could you also double-check the Unit Tests, and if we don't have one already, let's make sure to make a `BaseAnimationUnitTests.cs` class that every current + future `Animation` will inherit from where it automatically tests + verifies the length of the Animation. Otherwise, I can foresee future Animation PRs also missing this nuance too.

**Reply by bijington** [reply]
I have added this although it is somewhat flaky right now. I will keep playing around to determine why and what the best approach is:

https://github.com/CommunityToolkit/Maui/blob/163ed96dc040d67c20c0686227b17696840d3540/src/CommunityToolkit.Maui.UnitTests/Animations/BaseAnimationTests.cs

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
